source/extensions/access_loggers/file/config.cc

```diff
@@ -22,11 +23,24 @@ FileAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
   const auto& fal_config =
       MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::FileAccessLog&>(config);
   AccessLog::FormatterPtr formatter;
-  if (fal_config.format().empty()) {
-    formatter = AccessLog::AccessLogFormatUtils::defaultAccessLogFormatter();
+
+  if (fal_config.access_log_format_case() == envoy::config::accesslog::v2::FileAccessLog::kFormat ||
+      fal_config.access_log_format_case() ==
+          envoy::config::accesslog::v2::FileAccessLog::ACCESS_LOG_FORMAT_NOT_SET) {
+    if (fal_config.format().empty()) {
```

Comment:
Why is the logic for non-JSON different to before here?

Reply:
It's not strictly different, it has just been moved inside the check for 
` if (fal_config.access_log_format_case() == envoy::config::accesslog::v2::FileAccessLog::kFormat ||
      fal_config.access_log_format_case() ==
          envoy::config::accesslog::v2::FileAccessLog::ACCESS_LOG_FORMAT_NOT_SET) `

i.e:

If the `format` key was provided, or neither key was provided, fall back to the old logic
else if the `json_format` key was provided, use the json logic
else something bad happened, bail